### DISCUSSION TITLE
Post-process the results data from multiple designs in the same AEDT project?

### DISCUSSION THREAD

**Discussion by onyx22574** [ansys/pyaedt#4113] [Q&A]
I 'd like to use the integrated PyAEDT Jupyter Lab and read ANSYS Q3D simulation results. I am starting off the built-in PyAEDT from within ANSYS AEDT 2023R2 IDE environment and run Jupyter Notebook. 
I need to read the multiple results from optimetrics simulation where parametric sweep has been applied.
How to read out the results from AEDT Q3D project with multiple designs?
How to update (add) parameters in Optimetrics and re-run existing design and get the new data?

![image](https://github.com/ansys/pyaedt/assets/84866154/e26aec04-11fd-4a13-9430-195f50738cea)

**Comment by Samuelopez-ansys** [comment]
@onyx22574 Once you are connected to the AEDT session with the project opened. You can just connect to each Q3D design doing:

app1 = pyaedt.Q3d(designname=...)
app2 = pyaedt.Q3d(designname=...)


Then if I understood correctly, on each Q3D design, you have different optimetrics sweeps solved. You can use get_solution_data method to get all data, and then the object returned should contain all the needed variations.

You can add new variations like this example:

https://aedt.docs.pyansys.com/version/stable/examples/01-Modeling-Setup/Optimetrics.html#sphx-glr-examples-01-modeling-setup-optimetrics-py

**Comment by LuWan-dot** [comment]
Hi all,
I found the same issue here. In the API, there is a function called export_matrix_data. 
However, it seems I can only get access the nominal solution in the Analysis-Setup 1. This function cannot get results of other parametric settings, such as the results in Optimetrics (ParametricSetup).

for instance, 
this code works, which can extract the nominal solution
solution_data = app.export_matrix_data("D:\\xx\\xx\\test4.m",problem_type="AC RL",setup="Setup1")

However, it doesn't work if we specify the design variation:
solution_data = app.export_matrix_data("D:\\xx\\xx\\test4.m",problem_type="AC RL",setup="Setup1",variations ="$CreepDistance='0.8mm' $HeightGateCylinder1='1.3mm' )

**Comment by gmalinve** [comment]
@onyx22574 please refer to the answer I gave to @LuWan-dot in this discussion: https://github.com/ansys/pyaedt/discussions/6014#discussioncomment-12794793.
